File: easy/1582_special_positions_in_a_binary_matrix/solution.py
from typing import List


class Solution:
    def numSpecial(self, matrix: List[List[int]]) -> int:
        rows = len(matrix)
        cols = len(matrix[0])

        row_sums = []
        col_sums = [0] * cols

        for row in range(rows):
            row_sums.append(sum(matrix[row]))

            for col in range(cols):
                col_sums[col] += matrix[row][col]

        count = 0

        for row in range(rows):
            for col in range(cols):
                if matrix[row][col] == row_sums[row] == col_sums[col] == 1:
                    count += 1

        return count

File: easy/1582_special_positions_in_a_binary_matrix/solution.cpp
using namespace std;

#include <vector>

class Solution {
public:
  int numSpecial(vector<vector<int>> &matrix) {
    const int rows = matrix.size();
    const int cols = matrix[0].size();

    vector<int> rowSums(rows, 0);
    vector<int> colSums(cols, 0);

    for (int row = 0; row < rows; ++row) {
      for (int col = 0; col < cols; ++col) {
        if (matrix[row][col] == 1) {
          rowSums[row] += matrix[row][col];
          colSums[col] += matrix[row][col];
        }
      }
    }

    int count = 0;

    for (int row = 0; row < rows; ++row) {
      if (rowSums[row] == 1) {
        for (int col = 0; col < cols; ++col) {
          if (colSums[col] == 1 && matrix[row][col] == 1)
            ++count;
        }
      }
    }

    return count;
  }
};
